src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -84,8 +91,11 @@ public Result parseFile(final Path path) throws IOException {
     }
     List<String> fileLines = Arrays.asList(file.split(fileLineEnding.getChars()));
     ParseResult<CompilationUnit> parseResult = new JavaParser().parse(file);
-    CompilationUnit unit =
-        parseResult.getResult().orElseThrow(() -> new IOException("Unable to parse " + path));
+    Optional<CompilationUnit> unitOptional = parseResult.getResult();
+    if (!parseResult.isSuccessful() || !unitOptional.isPresent()) {
+      throw new IOException("Unable to parse " + path);
+    }
+    CompilationUnit unit = unitOptional.get();
```

Comment:
I'd have to check the parser code, but I'm pretty sure the additional check to see if the parse is successful is redundant. It was successful already if the Optional contained a value. Pretty sure this can go back to what it was.

Reply:
I have a test case in the test file I created. The test case checks if an `IOException` is thrown.

Without this check, there is no error or anything (maybe it just leaves the file as-is).

The `Optional` has a value but the content seems to be set to something like `???` instead of Java code.

Reply:
So you might expect in the case of a syntax error that the `Optional` is not present, but that does not seem to be the case.

Reply:
I looked at the docs and it says `Returns: the AST of the parsed source code, or empty if parsing failed completely.` So, if it can parse some, but not completely, you will get a partial `Result` and not an empty `Optional`.